src/alignment.js

```diff
@@ -85,8 +85,35 @@ var alignWithDOM = function(nodeName, key, statics) {
 };
 
 
+/**
+ * Clears out any unvisited Nodes, as the corresponding virtual element
+ * functions were never called for them.
+ * @param {!Element} node
+ */
+var clearUnvisitedDOM = function(node) {
+  var data = getData(node);
+  var lastChild = node.lastChild;
```

Comment:
For me, the jsperf is less than a 0.5% speed difference. This is part of the library is rarely hit and is not a bottleneck in the performance.

I prefer the existing one since it is easier for me to read and reason about. The shorter a function and the simpler constructs it uses, the is the easier it is to fit the whole thing in my working memory.

I'm not a neuroscientist, but from my own experience, using shorter functions and using simpler constructs like while() {} rather than do {} while() makes it much easier for me to focus on the important things like making sure all edge cases are handled and performance where it matters.

Everyone is different, but for me, after about 5-6 statements into a function I can't really process what is going on. Functions over 50 lines of code are entirely impossible for me to understand, I just give up after a few minutes of trying to understand what is going on.

I almost always prefer shorter functions/code in general and I don't see a tiny potential performance different really being worth restructuring the code.